include/eve/module/real/core/function/regular/simd/x86/convert_aggregated.hpp

```diff
@@ -0,0 +1,82 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/as.hpp>
+#include <eve/concept/value.hpp>
+#include <eve/detail/implementation.hpp>
+#include <eve/traits/as_floating_point.hpp>
+
+namespace eve::detail
+{
+  //================================================================================================
+  // Converting from aggregation of x86_128 to x86_128
+  //================================================================================================
+  template<real_scalar_value In, typename N, real_scalar_value Out>
+  requires( std::same_as<typename wide<Out,N>::abi_type, x86_128_>)
+  EVE_FORCEINLINE wide<Out,N>
+  convert_(EVE_SUPPORTS(sse2_), wide<In, N, aggregated_> const &v0, as_<Out> const &tgt) noexcept
+  {
+    // shortx16 -> charx16 is the only case we care about
+    if constexpr( std::is_integral_v<In> && (sizeof(In) == 2) )
+    {
+      auto[l,h] = v0.slice();
+      l &= static_cast<In>(0x00FFU);
+      h &= static_cast<In>(0x00FFU);
+
+      return _mm_packus_epi16(l,h);
+    }
+    else if constexpr( std::is_integral_v<In> && (sizeof(In) == 4) )
+    {
+      if constexpr(sizeof(Out) == 2)
+      {
+        auto[l,h] = v0.slice();
+
+        if constexpr( current_api >= sse4_1 )
+        {
+          l &= static_cast<In>(0x0000FFFFU);
+          h &= static_cast<In>(0x0000FFFFU);
+
+          return _mm_packus_epi32(l,h);
+        }
+        else if constexpr( current_api >= ssse3 )
+        {
+          auto l16 = bit_cast(l, as_<wide<Out>>());
+          auto h16 = bit_cast(h, as_<wide<Out>>());
+
+          return l16[pattern<0,2,4,6,-1,-1,-1,-1>] | h16[pattern<-1,-1,-1,-1,0,2,4,6>];
```

Comment:
Why not just call the propr intrinsic?
I don't think we should use engage all of the insane shuffle machinery internally unless we have to (like in sort or smth).